### PR TITLE
Add partner, host fields to Department

### LINKED ISSUE #2133: Feature - All Users / Partner Department Data & Display
# Description

**Is your feature request related to a problem? Please describe.**

Right now, the list of partner departments on the site has to be updated by a dev in the database, and while this updates the list in most places on the site, it doesn't update it on the FAQ, which results in multiple updates being made by (sometimes) multiple devs to ensure everything is in sync.

**Describe the solution you'd like**

We have an existing admin UI for departments, but it lacks the ability to add them.
Ideally, we'd upgrade this UI to do the following:
- add/edit/delete departments
  - this means the ability to add both english and french names
- check off, per department, whether that department is a Talent Cloud partner
- add both english and french impact statements

Once this work is done, we'd be able to populate the FAQ with this list in both languages, removing the extra step of updating the content manually.

**Describe alternatives you've considered**

@laurenoncoffee mentioned that if it's too complicated to have the partner checkbox described above, the entire list can be partners, and it would be okay to be able to simply delete them from the list.

### PR DESCRIPTION
**Pull request by mnigh**
Resolves #2133

### CODE REVIEW

File: app/Http/Controllers/Admin/DepartmentCrudController.php [hunk 1]
```diff
@@ -108,6 +121,32 @@ public function setupListOperation()
...
+            'type' => 'check',
+            'label' => 'Talent Cloud Host',
+        ]);
+
+        // Add filter for departments that are partners
+        $this->crud->addFilter(
+            [
+                'type' => 'simple',
+                'name' => 'partners',
+                'label'=> 'Department Partners'
```
Comment (Khristophor) [inline app/Http/Controllers/Admin/DepartmentCrudController.php:142]: Pedantic change, but we refer to them as "Partner Departments"

**Review by Khristophor — CHANGES_REQUESTED** [review]

**Review by Khristophor — APPROVED** [review]
Looks good. We'll need to make sure people are aware that Departments need to be tagged and ordered correctly when this change is deployed in order to have the FAQ page display the correct info.

**Comment by mnigh** [comment]
> Looks good. We'll need to make sure people are aware that Departments need to be tagged and ordered correctly when this change is deployed in order to have the FAQ page display the correct info.

@Khristophor Indeed, is there a formal place to put these instructions for a post-release?

**Comment by Khristophor** [comment]
> > Looks good. We'll need to make sure people are aware that Departments need to be tagged and ordered correctly when this change is deployed in order to have the FAQ page display the correct info.
> 
> @Khristophor Indeed, is there a formal place to put these instructions for a post-release?

Good question. @gobyrne do we have anywhere (aside from Slack) to notify people about things like this before/after deployment?

**Comment by gobyrne** [comment]
> > > Looks good. We'll need to make sure people are aware that Departments need to be tagged and ordered correctly when this change is deployed in order to have the FAQ page display the correct info.
> > 
> > 
> > @Khristophor Indeed, is there a formal place to put these instructions for a post-release?
> 
> Good question. @gobyrne do we have anywhere (aside from Slack) to notify people about things like this before/after deployment?

If you are just notifying, slack seems appropriate. Is there something else we need?

**Comment by tristan-orourke** [comment]
We don't have a good way of reminding _ourselves_ that something needs to be done when we deploy. Maybe adding a note in `deploy-instructions.tsx`? I always double check that file before deploying.

**Comment by mnigh** [comment]
> Maybe adding a note in `deploy-instructions.tsx`?
@tristan-orourke that would make sense if it was an instruction that was recurring and therefore going to be committed to the repository, whereas this case is a one off action that is required. is there somewhere in comments or notes for the next release that would be suited for this case?

**Comment by Khristophor** [comment]
> > Maybe adding a note in `deploy-instructions.tsx`?
> 
> @tristan-orourke that would make sense if it was an instruction that was recurring and therefore going to be committed to the repository, whereas this case is a one off action that is required. is there somewhere in comments or notes for the next release that would be suited for this case?

We don't have a changelog or similar, so it might make sense to put in the deploy-instructions file as a `this release only` note and remove in the future. @tristan-orourke ?

**Comment by tristan-orourke** [comment]
> We don't have a changelog or similar, so it might make sense to put in the deploy-instructions file as a `this release only` note and remove in the future. @tristan-orourke ?

@Khristophor Yeah, that's what I was thinking. I don't think there's a better place that we're used to checking, unfortunately.

We do use github to build a changelog from all the PRs that have been merged since last release, but its going to be reaaaally long, and we're not in the habit of looking through it before deployment.

**Comment by mnigh** [comment]
@tristan-orourke I have added some temporary instructions to the [deploy-instructions.txt file](https://github.com/GCTC-NTGC/TalentCloud/pull/4592/commits/64e9f7378ae5a05ad05f24e7bf520ee9ad3728e5). If this is clear enough, please feel free to merge this into the dev branch, thanks.

**Pull request merged**